### PR TITLE
deps: update Abseil library

### LINKED ISSUE #13973: compile_time_options CI job fails with the latest Abseil lib
*Title*: *compile_time_options CI job fails with the latest Abseil lib*

*Description*:
The int128 support in SimpleAtoi introduced in https://github.com/abseil/abseil-cpp/commit/4b915e70929ca2d6152240facc83d3d38c5d5f28 triggers this bug in Clang: [__int128_t multiply with -fsanitize=undefined fails to link (undefined reference to __muloti4)](https://bugs.llvm.org/show_bug.cgi?id=16404)

*Repro steps*:
1. update Abseil to the latest master in `bazel/repository_locations.bzl`;
2. run `ci/run_envoy_docker.sh 'ci/do_ci.sh bazel.compile_time_options'`

*Logs*:
The result can be seen in this CI job's log https://dev.azure.com/cncf/envoy/_build/results?buildId=56675&view=logs&jobId=e969334a-0e55-5c18-ac96-8b546753391e&j=e969334a-0e55-5c18-ac96-8b546753391e&t=b5329a9e-93c7-5412-9f4c-6f8d659e1b57
```
2020-11-05T15:34:46.1334065Z ERROR: /build/tmp/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/external/envoy/test/integration/BUILD:254:14: Linking of rule '@envoy//test/integration:hotrestart_test' failed (Exit 1): clang failed: error executing command 
2020-11-05T15:34:46.1335612Z   (cd /build/tmp/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/execroot/envoy && \
2020-11-05T15:34:46.1336305Z   exec env - \
2020-11-05T15:34:46.1336703Z     BAZEL_DO_NOT_DETECT_CPP_TOOLCHAIN=1 \
2020-11-05T15:34:46.1337306Z     BAZEL_LINKLIBS=-l%:libstdc++.a \
2020-11-05T15:34:46.1337873Z     BAZEL_LINKOPTS=-lm \
2020-11-05T15:34:46.1338251Z     CC=clang \
2020-11-05T15:34:46.1338572Z     CXX=clang++ \
2020-11-05T15:34:46.1338900Z     ENVOY_ASAN=1 \
2020-11-05T15:34:46.1339237Z     ENVOY_UBSAN_VPTR=1 \
2020-11-05T15:34:46.1339644Z     PATH=/usr/sbin:/usr/bin:/sbin:/bin:/opt/llvm/bin \
2020-11-05T15:34:46.1340034Z     PWD=/proc/self/cwd \
2020-11-05T15:34:46.1340759Z   /opt/llvm/bin/clang @bazel-out/k8-dbg/bin/external/envoy/test/integration/hotrestart_test-2.params)
2020-11-05T15:34:46.1341345Z Execution platform: @rbe_ubuntu_clang//config:platform
2020-11-05T15:34:46.1341798Z ld.lld: error: undefined symbol: __muloti4
2020-11-05T15:34:46.1342380Z >>> referenced by int128_have_intrinsic.inc:251 (external/com_google_absl/absl/numeric/int128_have_intrinsic.inc:251)
2020-11-05T15:34:46.1343451Z >>>               bazel-out/k8-dbg/bin/external/com_google_absl/absl/strings/_objs/strings/numbers.pic.o:(absl::operator*(absl::int128, absl::int128))
2020-11-05T15:34:46.1344374Z clang-10: error: linker command failed with exit code 1 (use -v to see invocation)
```

### PR DESCRIPTION
**Pull request by rojkov**
Commit Message: deps: update Abseil library
Additional Description: Also enforce linking with libc++ when sanitizers are enabled to resolve successfully symbols missing from gcc's c++ runtime.
Risk Level: Low
Testing: run integration tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A
Fixes #13973

### CODE REVIEW

**Comment by rojkov** [comment]
@lizan It seems the `clang-asan` build can't link properly with libc++ from LLVM 7.0.1. If I drop `--define boringssl=fips` from `COMPILE_TIME_OPTIONS` all the tests pass in the `linux_x64 compile_time_options` check.

Can we drop `--define boringssl=fips`?

File: ci/do_ci.sh [hunk 1]
```diff
@@ -304,7 +304,7 @@ elif [[ "$CI_TARGET" == "bazel.compile_time_options" ]]; then
 
   # Legacy codecs "--define legacy_codecs_in_integration_tests=true" should also be tested in
   # integration tests with asan.
-  bazel_with_collection test "${BAZEL_BUILD_OPTIONS[@]}" --define wasm=wavm "${COMPILE_TIME_OPTIONS[@]}" -c dbg @envoy//test/integration/... --config=clang-asan --build_tests_only
```
Comment (lizan) [inline ci/do_ci.sh:307]: actually this test does nothing... will be removed in https://github.com/envoyproxy/envoy/pull/14105

**Comment by ggreenway** [comment]
> @lizan It seems the `clang-asan` build can't link properly with libc++ from LLVM 7.0.1. If I drop `--define boringssl=fips` from `COMPILE_TIME_OPTIONS` all the tests pass in the `linux_x64 compile_time_options` check.
> 
> Can we drop `--define boringssl=fips`?

No. This is the only CI coverage to ensure that `fips` builds.

**Review by lizan — APPROVED** [review]

**Pull request merged**